packages/algob/src/internal/deployer.ts

```diff
@@ -742,33 +744,62 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
    * https://developer.algorand.org/docs/features/asc1/stateless/sdks/#account-delegation-sdk-usage
    * @param name: Logic Signature filename (must be present in assets folder)
    * @param signer: Signer Account which will sign the smart contract
+   * @param lsigName name of smart signature (checkpoint info will be stored against this name)
    * @param scTmplParams: scTmplParams: Smart contract template parameters
    *     (used only when compiling PyTEAL to TEAL)
-   * @param lsigName name of smart signature (if passed, checkpoint info will be stored against this name)
    */
   async mkDelegatedLsig (
     fileName: string,
+    lsigName: string,
     signer: rtypes.Account,
-    scTmplParams?: SCParams,
-    lsigName?: string
+    scTmplParams?: SCParams
   ): Promise<LsigInfo> {
     return await this._mkLsig(fileName, scTmplParams, signer, lsigName);
   }
 
   /**
    * Stores logic signature info in checkpoint for contract mode
    * @param fileName ASC file name
+   * @param lsigName name of lsig (checkpoint info will be stored against this name)
    * @param scTmplParams: Smart contract template parameters (used only when compiling PyTEAL to TEAL)
-   * @param lsigName name of lsig (if passed, checkpoint info will be stored against this name)
    */
   async mkContractLsig (
     fileName: string,
-    scTmplParams?: SCParams,
-    lsigName?: string
+    lsigName: string,
```

Comment:
if we are moving this parameter, then let's put it as the first argument.  Also need to update changelog

Reply:
ok